api/tests/unit/features/test_unit_features_views.py

```diff
@@ -2554,3 +2559,66 @@ def test_simple_feature_state_returns_only_latest_versions(
 
     response_json = response.json()
     assert response_json["count"] == 2
+
+
+@pytest.mark.freeze_time(two_hours_ago)
+def test_feature_list_last_modified_values(
+    staff_client: APIClient,
+    staff_user: FFAdminUser,
+    environment_v2_versioning: Environment,
+    project: Project,
+    feature: Feature,
+    with_project_permissions: WithProjectPermissionsCallable,
+    django_assert_num_queries: DjangoAssertNumQueries,
+) -> None:
+    # Given
+    # another v2 versioning environment
+    environment_v2_versioning_2 = Environment.objects.create(
+        name="environment 2", project=project, use_v2_feature_versioning=True
+    )
+
+    url = "{base_url}?environment={environment_id}".format(
+        base_url=reverse("api-v1:projects:project-features-list", args=[project.id]),
+        environment_id=environment_v2_versioning.id,
+    )
+
+    with_project_permissions([VIEW_PROJECT])
+
+    with freeze_time(one_hour_ago):
+        # create a new published version in another environment, simulated to be one hour ago
+        environment_v2_versioning_2_version_2 = (
+            EnvironmentFeatureVersion.objects.create(
+                environment=environment_v2_versioning_2, feature=feature
+            )
+        )
+        environment_v2_versioning_2_version_2.publish(staff_user)
+
+    with freeze_time(now):
+        # and create a new unpublished version in the current environment, simulated to be now
+        # this shouldn't affect the values returned
+        EnvironmentFeatureVersion.objects.create(
+            environment=environment_v2_versioning, feature=feature
+        )
+
+    # let's add a few more features to ensure we aren't adding N+1 issues
+    for i in range(2):
+        Feature.objects.create(name=f"feature_{i}", project=project)
+
+    # When
+    with django_assert_num_queries(14):  # TODO: reduce this number of queries!
```

Comment:
Yeah, that crossed my mind as soon as I saw the annotate + prefetch related combo.

Reply:
Just for a sanity check, I commented out all the annotations and re-ran the test which confirmed that I haven't actually added any queries here (since the `prefetch_related` call hasn't actually changed).